digitalfabrik/entitlementcard
# 1571: Import stores backend

### Short description

As a user i want to import stores accepting stores via csv. The backend part is implemented here

### Proposed changes



- create reusable functions for mapping and storing the accepting store data in the database
- add a dryRun function to check how many stores are affected, before applying db actions
- add missing react component tests
- add missing tests for backend functions
- TODO integration test for the mutation (#1605), since i couldn't get mocking bearer token for and have to investigate more

### Side effects



- the normal store import function may be affected due to the refactoring. Please test if the standard store import (runConfig) still works. Here you can find a csv with the correct scheme. 
https://github.com/digitalfabrik/sozialpass-nuernberg-data/blob/main/nuernberg-akzeptanzstellen.csv
Note: If you have added the koblenz project to your `config.local.yml` you have to set this
`pipelineName: BerechtigungskarteShowcase` for koblenz to skip the store import via pipeline

### Testing


1. Create a Project Store Manager and login
2. Go to "Akzeptanzpartner verwalten"
3. Add a csv, data can be found here (https://github.com/digitalfabrik/entitlementcard/pull/1572)
4. Import the stores and check the database tables
5. Open native nuernberg app and check if the stores and the correct data is available
6. Do the same using "testlauf" flag with a different csv and check that the database was not affected

### Resolved issues



Fixes: #1571
